cjapy/cjapy.py

```diff
@@ -2560,3 +2560,256 @@ def clean_column_name(col):
+            removeSingleEventPeople : OPTIONAL : Removes people that had only a single event in the date range.
+            filterId : OPTIONAL : The segment ID to filter the report with.
+        
+        Returns:
+            A pandas DataFrame containing the person profile data.
+        """
+
+        if featureMetrics is None:
+            raise ValueError("Require at least one feature metric")
+        if dataviewId is None:
+            raise ValueError("Require a Data View ID")
+        def ensure_dimension_prefix(dimension: str) -> str:
+            if not dimension.startswith("variables/"):
+                return f"variables/{dimension}"
+            return dimension
+        def ensure_metric_prefix(metric: str) -> str:
+            if not metric.startswith("metrics/"):
+                return f"metrics/{metric}"
+            return metric
+        def ensure_datetime_format(date: Union[str, datetime], is_start: bool) -> str:
+            if isinstance(date, str):
+                date = datetime.fromisoformat(date)
+            if date.time() == datetime.min.time():
+                if is_start:
+                    return date.strftime('%Y-%m-%dT00:00:00.000')
+                else:
+                    return date.strftime('%Y-%m-%dT23:59:59.999')
+            return date.strftime('%Y-%m-%dT%H:%M:%S.000')
+        if startDate is None:
+            startDate = datetime.today() - timedelta(days=90)
+        if endDate is None:
+            endDate = datetime.today()
+        start_date_str = ensure_datetime_format(startDate, is_start=True)
+        end_date_str = ensure_datetime_format(endDate, is_start=False)
+        date_range = f"{start_date_str}/{end_date_str}"
+        personId = ensure_dimension_prefix(personId)
+        metrics = [ensure_metric_prefix(metric) for metric in featureMetrics]
+        targetMetric = ensure_metric_prefix(targetMetric)
+        
+        dataRequest = RequestCreator()
+        dataRequest.setDataViewId(dataviewId)
+
+        sampling_metric_definition = {
+            "func": "calc-metric",
+            "formula": {
+                "func": "add",
+                "col1": {
+                "func": "modulo",
+                "description": "Modulo",
+                "col1": {
+                    "func": "add",
+                    "col1": {
+                    "func": "multiply",
+                    "col1": {
+                        "func": "visualization-group",
+                        "col": {
+                        "func": "add",
+                        "col1": {
+                            "func": "cumul",
+                            "description": "Cumulative",
+                            "n": 0,
+                            "col": 1
+                        },
+                        "col2": sampleSeed
+                        }
+                    },
+                    "col2": 1664525
```

Comment:
Are we sure about having fixed value ? 
I am always thinking that hardcoded value may be a typo so just asking.

Reply:
https://en.wikipedia.org/wiki/Linear_congruential_generator

The hard-coded values are common values used in the LCG random number generator. These values are used to create a random sample of users from CJA to pull from.